repository-data/webfiles/src/main/resources/site/freemarker/hstdefault/essentials-carousel.ftl

```diff
@@ -0,0 +1,108 @@
+<#if pageable?? && pageable.items?has_content>
+    <#if cparam.pause>
+        <#assign pauseCarousel = 'hover'/>
+    <#else>
+        <#assign pauseCarousel = ''/>
+    </#if>
+    <div id="${componentId}" class="carousel slide" data-ride="carousel" data-interval="${cparam.interval?c}"
+         data-pause="${pauseCarousel}" data-wrap="${cparam.cycle?string}">
+        <ol class="carousel-indicators">
+            <#list 0..(pageable.total-1) as index>
+                <#if index==0>
+                    <li data-target="#${componentId}" data-slide-to="${index}" class="active"></li>
+                <#else>
+                    <li data-target="#${componentId}" data-slide-to="${index}"></li>
+                </#if>
+            </#list>
+        </ol>
+        <div class="carousel-inner">
+            <#list pageable.items as item>
+                <#if item_index==0>
+                    <#assign active = ' active'/>
+                <#else>
+                    <#assign active = ''/>
+                </#if>
+                <div class="item${active}">
+                  <@hst.manageContent hippobean=item/>
+                  <img src="<@hst.link hippobean=item.image />" alt="${item.title}"/>
+                    <div class="carousel-caption">
+                        <#if item.link??>
+                            <h3><a href="<@hst.link hippobean=item.link/>">${item.title}</a></h3>
+                        <#else>
+                            <h3>${item.title}</h3>
+                        </#if>
+                        <@hst.html hippohtml=item.content/>
+                    </div>
+                </div>
+            </#list>
+        </div>
+        <#if cparam.showNavigation>
+            <a class="left carousel-control" href="#${componentId}" data-slide="prev"><span class="glyphicon glyphicon-chevron-left"></span></a>
+            <a class="right carousel-control" href="#${componentId}" data-slide="next"><span class="glyphicon glyphicon-chevron-right"></span></a>
+        </#if>
+        <#--
+          The Carousel component is initialized on page-load by means of the data attributes. However, when the
+          channel-manager redraws a container (after actions like adding, removing or reordering components) it will only
+          do a page reload if one of the affected components adds a headContribution that has not been processed yet
+          (see HSTTWO-3747). To ensure it is also initialiazed when the page is *not* reloaded, the following snippet is
+          used.
+        -->
+        <#if editMode>
+          <script type="text/javascript">
+            if (window.jQuery && window.jQuery.fn.carousel) {
+              jQuery('#${componentId}').carousel();
+            }
+          </script>
+        </#if>
+    </div>
+
+    <@hst.headContribution category="htmlHead">
+        <style type="text/css">
+            /* Carousel base class */
```

Comment:
Shouldn't this all be in CSS/SASS files instead of inline?

Reply:
@robValtech

Reply:
I don't know much about the carousel component, but as I can see the inline CSS uses some Freemarker variables to construct the styling definitions, so I reckon it is best to keep it as it.

Reply:
Ah good point, will leave it then